pkg/gcpcredential/gcpcredential.go

```diff
@@ -259,6 +260,15 @@ func (g *ContainerRegistryProvider) Provide(image string) credentialconfig.Docke
 		Email:    string(email),
 	}
 
+	// If UseRegistryFromImage is true, we won't examine if it's matching containerRegistryUrls.
+	// Currently, this is only used by auth-provider-gcp.
+	if g.UseRegistryFromImage {
+		if registry, _, found := strings.Cut(image, "/"); found {
+			cfg[registry] = entry
+		}
+		return cfg
+	}
+
 	// Add our entry for each of the supported container registry URLs
 	for _, k := range containerRegistryUrls {
```

Comment:
I don't remember the logic too well. But here is the question. Old logic added all wildcarded urls. My thinking here - if image url will redirect from one gcr.io blob to another, we will still use the auth token. With the new approach, no redirects will have the auth token.

Is it a real issue that may happen? Who would be the best person to review it from security perspective?

Reply:
>  if image url will redirect from one gcr.io blob to another, we will still use the auth token. With the new approach, no redirects will have the auth token

I think the change for the new logic is that we don't have allowed endpoints for registry anymore(line 55), instead, we will get the token no matter what the registry is in the image url. I don't fully sure what the re-direct is in this case.

Reply:
If the redirect happens within the container runtime pull attempt, it doesn't have a chance to go get another credential for another registry... I don't actually know what happens if the container runtime is asked to pull an image from registry `foo` using a credential config like `{"foo":{...}}` and the registry redirects to registry `bar` ... does the container runtime look for an entry in the credential matching `bar` or reuse the entry for `foo`?

Reply:
@liggitt you are right, it will be on container runtime side. Auth will only be reused for the same domain

Reply:
is there a way to return a wildcard to indicate whatever registry the request gets redirected to can use the credential? Is that a reasonable thing to do?